### PR TITLE
provide multiple way to access the size of the environment

### PR DESCRIPTION
**Pull request by irevoire**
# Pull Request

## What does this PR do?
Provide multiple ways to access to the size of the databases.
Also provide a getter on the flags.

### CODE REVIEW

File: heed/src/env.rs [hunk 1]
```diff
@@ -276,8 +276,48 @@ pub enum CompactionOption {
...
+        Ok(path.metadata()?.len())
+    }
+
+    /// Check if a flag was specified when opening the environment.
+    pub fn contain_flag(&self, flag: Flags) -> Result<bool> {
+        let flags = self.raw_flags()?;
+        println!("{flags:032b} &");
+        println!("{:032b}", flag as u32);
+        let or = flags & (flag as u32);
+        println!("{or:032b}");
```
Comment (Kerollmops) [inline heed/src/env.rs:296]:
```suggestion
        let flags = self.raw_flags()?;
        let or = flags & (flag as u32);
```

File: heed/src/env.rs [hunk 2]
```diff
@@ -276,8 +276,48 @@ pub enum CompactionOption {
...
+            self.path().to_path_buf()
+        } else {
+            self.path().join("data.mdb")
+        };
+
+        Ok(path.metadata()?.len())
+    }
+
+    /// Check if a flag was specified when opening the environment.
+    pub fn contain_flag(&self, flag: Flags) -> Result<bool> {
```
Comment (Kerollmops) [inline heed/src/env.rs:291]:
```suggestion
    pub fn contains_flag(&self, flag: Flags) -> Result<bool> {
```

File: heed/src/env.rs [hunk 3]
```diff
@@ -276,8 +276,48 @@ pub enum CompactionOption {
...
+        let flags = self.raw_flags()?;
+        println!("{flags:032b} &");
+        println!("{:032b}", flag as u32);
+        let or = flags & (flag as u32);
+        println!("{or:032b}");
+
+        Ok(or != 0)
+    }
+
+    /// Return the raw flags stored the environment was opened with.
```
Comment (Kerollmops) [inline heed/src/env.rs:301]: I don't understand this sentence.

File: heed/src/env.rs [hunk 4]
```diff
@@ -276,8 +276,48 @@ pub enum CompactionOption {
...
+    pub fn raw_flags(&self) -> Result<u32> {
+        let mut flags = std::mem::MaybeUninit::uninit();
+        unsafe { mdb_result(ffi::mdb_env_get_flags(self.env_mut_ptr(), flags.as_mut_ptr()))? };
+        let flags = unsafe { flags.assume_init() };
+
+        Ok(flags)
+    }
+
+    /// The map size that was set when configuring the environment.
+    pub fn map_size(&self) -> Result<usize> {
```
Comment (Kerollmops) [inline heed/src/env.rs:311]: I would say that the map size should be returned as an u64 instead, am I right?

**Review by Kerollmops — CHANGES_REQUESTED** [review]

File: heed/src/env.rs (hunk 3 above)
Comment (irevoire) [reply heed/src/env.rs]: Oops yes, changed to;
>  /// Return the raw flags the environment was opened with.

File: heed/src/env.rs (hunk 4 above)
Comment (irevoire) [reply heed/src/env.rs:308]: The ffi returns it as an `usize` so I kept it like that.

**Review by Kerollmops — APPROVED** [review]
Thank you very much! Will definitely be helpful!
Would it be possible to fix the tests, please?

**Review by Kerollmops — APPROVED** [review]
bors merge

**Pull request merged**